langchain/document_loaders/parsers/text_translate.py

```diff
@@ -0,0 +1,29 @@
+"""Module contains common parsers for PDFs."""
+from typing import Any, Iterator, Mapping, Optional, Union
+
+from langchain.document_loaders.base import BaseBlobParser
+from langchain.utils import get_from_env
+from langchain.document_loaders.blob_loaders import Blob
+from langchain.schema import Document
+
+
+class DoctranTranslateParser(BaseBlobParser):
+    """Translates text documents into other languages using doctran."""
+
+    def __init__(self, language: str, openai_api_key: Optional[str] = None):
+        self.openai_api_key = openai_api_key
+        self.language = language
+        try:
+            from doctran import Doctran, ExtractProperty
+            self.doctran = Doctran(openai_api_key=openai_api_key)
+        except ImportError:
+            raise ImportError("Install doctran to use this parser. (pip install doctran)")
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Lazily parse the blob."""
+        if self.openai_api_key:
+            openai_api_key = self.openai_api_key
+        else:
+            openai_api_key = get_from_env("openai_api_key", "OPENAI_API_KEY")
+        doctran_doc = self.doctran.parse(content=blob.as_string()).translate(language=self.language).execute()
```

Comment:
I think you need to pass `openai_api_key` here.

Reply:
Same fix as above, openai_api_key is passed into doctran constructor so I handled this all in the constructor for the parser